### ISSUE TITLE
Remove log4j and move to slf4j

### ISSUE THREAD

**Issue by michaelrice** [yavijava/yavijava#222]
It would be nice to remove the requirement for log4j with out loosing any of the benefits of logging, and still allowing the use of log4j for those that use it, while still opening the doors for others not using it to use what they already use (maybe java.util.logging or logback etc..)

**Comment by alex-lata** [comment]
+1

**Comment by mbap** [comment]
Hey Michael,

Thanks for fixing this issue. I just noticed this problem in a project of mine, and was curious if there was any plans to backport this into a bug fix release. The main issue with explicitly specifying which implementation is that if there are two dependencies and both are pulling different slf4j implementations you get into a bind, and one gets randomly chosen.

According to the docs I should kindly ask you. 

Thanks for your hard work.

```
Embedded components such as libraries or frameworks should not declare a dependency 
on any SLF4J binding but only depend on slf4j-api. When a library declares a compile-time
dependency on a SLF4J binding, it imposes that binding on the end-user, thus negating 
SLF4J's purpose. When you come across an embedded component declaring a 
compile-time dependency on any SLF4J binding, please take the time to contact 
the authors of said component/library and kindly ask them to mend their ways.
```

**Comment by michaelrice** [comment]
What exactly did I do wrong @mbap? If you want just send over a pull request with the fix. This change is only in the 6.5 branch right now anyway and Im not even 1/2 way done adding all the 6.5 stuff so there is plenty of time to get it fixed before there is a release.

**Comment by mbap** [comment]
Sorry if that came across as blaming you, that wasn't the intent. You did nothing wrong. 

What I'd like to know is how you plan/schedule bug fix releases, and if a7efb96 could be considered for another 6.0 bug fix release. I understand that the 6.5 stuff is a work in progress and that you already merged it into that development branch.

To clarify my original post, what I was trying to say is that I use yavijava in one of my projects, which has this log4j dependency. Log4j is an implementation of the slf4j implementation. The problem I have is that I have another dependency that pulls in a different slf4j implementation, specifically logback. What happens when both log4j and logback (2 different implementations of slf4j) are in the classpath is that slf4j will print warnings and then randomly pick one of implementations that are in the classpath. If there is no configuration for the randomly chosen logger implementation, then logging will just be a noop.

TLDR;; yavijava brings in log4j which might conflict with loggers of projects that depend on yavijava.

If your patch makes it into a sooner release than the 6.5 release, it would solve my issue, and hopefully others. I am happy to send in a patch if the work is incomplete, but it looks like you handled all of it in a7efb96.

**Comment by michaelrice** [comment]
Oh I see. I thought you were saying I might have done the wrong thing in the gradle file for how I declare the compile time dep. I do not plan to back port that work into a 6.0 branch because it changes how the lib would work. I do agree I shouldnt have done it this way to begin with by using log4j but I didnt know about slf4j when I did all the log4j stuff, and I think to change how the lib would work fundamentally like that is best done in a major release not a minor revision. 

I would be open to cutting a separate release labeled as like 6.0.x-slf4j Would that work? I guess if I did that I would have to be sure to patch 6.0.x and 6.0.x-slf4j with the same bug fixes going forward but that shouldnt be a huge problem..

**Comment by mbap** [comment]
Ya I agree with it being a larger amount of work, and your solution would definitely work for me. I def don't want to create a maintenance headache for you though, but if you feel it wouldn't be too tough that would be great.
